lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
@lundibundi

> now when we receive first half-message we concat it with null(initial state of class buffer)

Where does this happen exactly? I don't recall this behavior at all.

> upon each next half-message we receive

There literally can only be two half-messages :)

> then upon receiving consequent half-messages

Same here, can you please explain what you mean by "consequent half-messages"?

> This way we will only copy our accumulated data if there is not enough space in the class buffer (and we will grow it more that just the size of received data), hence avoiding a lot of copying in case of very fragmented data.

Yes and I was talking exactly about the growing part, it will lead to almost the same amount of copying we have now (see first paragraph in https://github.com/metarhia/jstp/pull/254#discussion_r128333507), and will lead to increased memory usage, I don't understand how this will be better for performance overall.